classes/plugins/DOIPubIdExportPlugin.inc.php

```diff
@@ -302,6 +302,54 @@ function getUnregisteredGalleys($context) {
 	}
 
 	/**
+	 * Get published articles with a DOI asigned from submission IDs.
+	 * @param $submissionIds array
+	 * @param $context Context
+	 * @return array
+	 */
+	function getPublishedArticles($submissionIds, $context) {
+		$publishedArticles = array();
+		$publishedArticleDao = DAORegistry::getDAO('PublishedArticleDAO');
+		foreach ($submissionIds as $submissionId) {
+			$publishedArticle = $publishedArticleDao->getPublishedArticleByArticleId($submissionId, $context->getId());
+			if ($publishedArticle && $publishedArticle->getStoredPubId('doi')) $publishedArticles[] = $publishedArticle;
+		}
+		return $publishedArticles;
+	}
+
+	/**
+	 * Get published issues with a DOI asigned  from issue IDs.
+	 * @param $issueIds array
+	 * @param $context Context
+	 * @return array
+	 */
+	function getPublishedIssues($issueIds, $context) {
+		$publishedIssues = array();
+		$issueDao = DAORegistry::getDAO('IssueDAO');
+		foreach ($issueIds as $issueId) {
+			$publishedIssue = $issueDao->getById($issueId, $context->getId());
+			if ($publishedIssue && $publishedIssue->getStoredPubId('doi')) $publishedIssues[] = $publishedIssue;
+		}
+		return $publishedIssues;
+	}
+
+	/**
+	 * Get article galleys with a DOI asigned  from gallley IDs.
+	 * @param $galleyIds array
+	 * @param $context Context
+	 * @return array
+	 */
+	function getArticleGalleys($galleyIds, $context) {
```

Comment:
@asmecher, this function overwrites a parent function. The parent function (in PubObjectsExportPlugin i.e. for export of published objects e.g. in DOAJ) returns all article galleys and this one (in DOIPubIdExportPlugin i.e. for DOI export plugins) returns only galleys with DOI. They are named the same because they are called at the same place in the same way, so that I don't have to differentiate there anything else. OK so?